src/Navigation.re

```diff
@@ -1,5 +1,11 @@
 type t;
 
+type navigationEventPayload = {
+  .
+  "type": string,
+  "key": string,
+};
+
```

Comment:
I meant

```reason
type navigationEventPayload = {
  .
  "action": NavigationActions.action,
  "context": string,
  "type": string,
  "state": NavigationState.t,
  "lastState": Js.Nullable.t(NavigationState.t),
}: 
```

and use that in `addListener` below instead of the inline type definition.